bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java

```diff
@@ -589,9 +576,43 @@ private void throwInvalidSubModuleError(Toml toml, Module module) {
         if (errorNode != null) {
             invalidRequiredModuleSet.add(module.toString());
             invalidTomlLines.add(errorNode.location().lineRange());
-            throw new ConfigException(CONFIG_TOML_INVALID_MODULE_STRUCTURE, getLineRange(errorNode), moduleName,
-                    moduleName);
+            throw new ConfigException(CONFIG_TOML_INVALID_MODULE_STRUCTURE_WITH_VARIABLE, getLineRange(errorNode),
+                    moduleName, variableName, moduleName);
+        }
+    }
+
+    private Optional<TomlNode> getErrorNode(Toml toml, String moduleName, String orgModuleKey, String variableName) {
+        Optional<TomlValueNode> valueNode = toml.get(moduleName);
+        Optional<Toml> tableNode = toml.getTable(moduleName);
+        List<Toml> tomlTables = toml.getTables(moduleName);
+        if (valueNode.isEmpty()) {
+            valueNode = toml.get(orgModuleKey);
+        }
+        if (tomlTables.isEmpty()) {
+            tomlTables = toml.getTables(orgModuleKey);
```

Comment:
Can we do early return here? tomlTables.isEmpty() is checking multiple places

Reply:
Same in tableNode.isEmpty() condition as well

Reply:
updated with early return statements.